stl/inc/ranges

```diff
@@ -1734,6 +1734,198 @@ namespace ranges {
+            return _Range;
+        }
+        _NODISCARD constexpr _Vw base() && noexcept(is_nothrow_move_constructible_v<_Vw>) /* strengthened */ {
+            return _STD move(_Range);
+        }
+
+        _NODISCARD constexpr const _Pr& pred() const noexcept /* strengthened */ {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Pred, "value-initialized take_while_view has no predicate");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+            return *_Pred;
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto begin() noexcept(
+            noexcept(_RANGES begin(_Range))) /* strengthened */ requires (!_Simple_view<_Vw>) {
+            // clang-format on
+            return _RANGES begin(_Range);
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto begin() const noexcept(
+            noexcept(_RANGES begin(_Range))) /* strengthened */ requires range<const _Vw>
+            && indirect_unary_predicate<const _Pr, iterator_t<const _Vw>> { // Per Resolution of LWG-3450
+            // clang-format on
+            return _RANGES begin(_Range);
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() noexcept(
+            noexcept(_RANGES end(_Range))) /* strengthened */ requires (!_Simple_view<_Vw>) {
+            // clang-format on
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Pred, "value-initialized take_while_view cannot call end");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+            return _Sentinel<false>{_RANGES end(_Range), _STD addressof(*_Pred)};
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto end() const noexcept(
+            noexcept(_RANGES end(_Range))) /* strengthened */ requires range<const _Vw>
+            && indirect_unary_predicate<const _Pr, iterator_t<const _Vw>> { // Per Resolution of LWG-3450
+            // clang-format on
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Pred, "value-initialized take_while_view cannot call end");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+            return _Sentinel<true>{_RANGES end(_Range), _STD addressof(*_Pred)};
+        }
+    };
+
+    template <class _Rng, class _Pr>
+    take_while_view(_Rng&&, _Pr) -> take_while_view<views::all_t<_Rng>, _Pr>;
+
+    namespace views {
+        // VARIABLE views::take_while
+        class _Take_while_fn {
+        private:
+            template <class _Pr>
+            struct _Partial : _Pipe::_Base<_Partial<_Pr>> {
+                /* [[no_unique_address]] */ _Semiregular_box<_Pr> _Pred;
+
+                template <viewable_range _Rng>
+                _NODISCARD constexpr auto operator()(_Rng&& _Range) const& noexcept(
+                    noexcept(take_while_view{_STD forward<_Rng>(_Range), *_Pred})) requires requires {
+                    take_while_view{static_cast<_Rng&&>(_Range), *_Pred};
+                }
+                {
+                    // clang-format on

```

Comment:
Ah I thought you did this to prevent it wrapping onto a single line `{ code here }`

Reply:
You give me too much credit - I think this was just a mistake. It would be weird to choose this particular comment to make the function body two lines long when any comment would suffice.